rules/enforce-single-end-return.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * @fileoverview Functions must have a single return at the end of the function.
+ * @author Mitchell Van Der Hoeff <mitchellvanderhoeff@gmail.com>
+ */
+
+'use strict';
+
+module.exports = {
+
+    meta: {
+
+        docs: {
+            recommended: false,
+            type: 'error',
+            description: 'Functions must have a single return at the end of the function.'
+        },
+
+        schema: []
+
+    },
+
+    create: function (context) {
+
+        function inspectFunctionDeclaration(emitted) {
+            var node = emitted.node;
+
+            if (emitted.exit) { return; }
+
+            let body = node.body.body;
+            if (body.length === 0) {
+                return;
+            }
+            for (let statement of body.slice(0, -1)) {
+                if (statement['type'] === 'ReturnStatement') {
+                    context.report({
+                        node: node,
+                        message: 'Return statement not at end of function',
```

Comment:
Change message to `"<FUNCTION NAME>": return statement must be at the end of the function only.`
You can get function name from the `node` object.

In the special case when function doesn't have a name (ie, the fallback function), message should be `Fallback function: return statement must be at the end of the function only.`